core/services/ocr2/plugins/mercury/plugin.go

```diff
@@ -55,8 +86,20 @@ func NewServices(
 	}
 	lggr = lggr.Named("MercuryPlugin").With("jobID", jb.ID, "jobName", jb.Name.ValueOrZero())
 
+	// encapsulate all the subservices and ensure we close them all if any fail to start
+	srvs := []job.ServiceCtx{ocr2Provider}
+	abort := func() {
+		if cerr := services.MultiCloser(srvs).Close(); err != nil {
+			lggr.Errorw("Error closing unused services", "err", cerr)
+		}
+	}
 	saver := ocrcommon.NewResultRunSaver(pipelineRunner, lggr, cfg.MaxSuccessfulRuns(), cfg.ResultWriteQueueDepth())
+	srvs = append(srvs, saver)
+
+	loopEnabled, loopCmd := env.MercuryPlugin.Cmd.Get() != "", env.MercuryPlugin.Cmd.Get()
```

Comment:
Maybe skip the double call?
```suggestion
	loopCmd := env.MercuryPlugin.Cmd.Get()
	loopEnabled := loopCmd != ""
```